File: Baekjoon/11723.cpp
#include <iostream>
using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	int n;
	cin >> n;
	int s = 0;
	while (n--) {
		string cmd;
		int num;
		cin >> cmd;

		if (cmd == "add") {
			cin >> num;
			s |= (1 << num);
		}
		else if (cmd == "remove") {
			cin >> num;
			s &= ~(1 << num);
		}
		else if (cmd == "check") {
			cin >> num;
			if (s & (1 << num)) cout << 1 << '\n';
			else cout << 0 << '\n';
		}
		else if (cmd == "toggle") {
			cin >> num;
			s ^= (1 << num);
		}
		else if (cmd == "all") {
			s = (1 << 21) - 1;
		}
		else if (cmd == "empty") {
			s = 0;
		}
	}
}

File: Baekjoon/11723.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    var s = 0
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    repeat(br.readLine().toInt()) {
        val cmd = br.readLine().split(" ")
        when {
            cmd[0] == "add" -> s = s or (1 shl cmd[1].toInt())
            cmd[0] == "remove" -> s = s and (-1 xor (1 shl cmd[1].toInt()))
            cmd[0] == "check" -> bw.write("${if (s and (1 shl cmd[1].toInt()) != 0) 1 else 0}\n")
            cmd[0] == "toggle" -> s = s xor (1 shl cmd[1].toInt())
            cmd[0] == "all" -> s = -1
            cmd[0] == "empty" -> s = 0
        }
    }
    bw.flush()
}